File: web/src/lib/payments.ts
import type { Enums } from "@/lib/database.types";

/**
 * To'lov integratsiyasi uchun joy (Payme / Click).
 *
 * Hozircha to'lov faqat `orders.payment_status` maydoni orqali modellashtirilgan:
 * pending → partial (bo'nak, ~30%) → paid.
 *
 * Haqiqiy integratsiyada bu modul provayder API'siga so'rov yuboradi va
 * webhook orqali payment_status'ni yangilaydi.
 */

export const DEPOSIT_RATE = 0.3;

export function depositAmount(totalPrice: number): number {
  return Math.round(totalPrice * DEPOSIT_RATE);
}

export function nextPaymentStatus(
  current: Enums<"payment_status">
): Enums<"payment_status"> | null {
  if (current === "pending") return "partial";
  if (current === "partial") return "paid";
  return null;
}

File: android/app/src/main/java/uz/tikuvchi/data/Payments.kt
package uz.tikuvchi.data

import uz.tikuvchi.data.model.PaymentStatus

/**
 * web/src/lib/payments.ts ning ko'chirmasi — to'lov integratsiyasi
 * (Payme/Click) uchun joy. Hozircha faqat payment_status maydoni orqali
 * modellashtirilgan: pending -> partial (bo'nak, ~30%) -> paid.
 */
object Payments {
    const val DEPOSIT_RATE = 0.3

    fun depositAmount(totalPrice: Long): Long =
        Math.round(totalPrice * DEPOSIT_RATE)

    fun nextPaymentStatus(current: PaymentStatus): PaymentStatus? = when (current) {
        PaymentStatus.PENDING -> PaymentStatus.PARTIAL
        PaymentStatus.PARTIAL -> PaymentStatus.PAID
        PaymentStatus.PAID -> null
    }
}
